### PR TITLE
GPWS Rework

### LINKED ISSUE #1335: [BUG] Landing with Flaps 3 should not trigger "too low, flaps" warning
**Mod Version**

latest master branch

**Describe the bug**

landing with flaps 3 is allowed, should not trigger "too low, flaps" warning

**To Reproduce**

1. during approach, set flaps to 3, and landing ( tested at CYPR)
2.
3.

**Expected behavior**

no warning

**Actual behavior**

triggers warning if not set flaps to FULL

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #1734: [BUG] "To Low terrain" on initial ground
**Mod Version**

0.4.1

**Describe the bug**

A non systematic bug: at start on parking, I have the vocal message "too low terrain" on repeat.
I see also, one moteur don't work even I repeat the ignition sequence.
To have a normal operation I had to click "Restart flight" menu and all works fine after.

**To Reproduce**

1.  Non systematic bug. The screenshot below shows the engine thrust #1 not in iddle position but the display shows no activity for it.

2.
3.

**Expected behavior**

Verify the startup sequence of script modd

**Actual behavior**

sometimes there are malfunctions from the start on the ground

**References**

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

**BY REMOVING THIS LINE, I ATTEST THAT I HAVE FOLLOWED THE ISSUE TEMPLATE AND PROVIDED A TITLE, as well as searched for other possible duplicate issues. I acknowledge if this line is not removed and/or the issue template is not respected, my issue will be closed without warning**

### LINKED ISSUE #1756: [BUG] Sink Rate Alarms in loop
**Mod Version**
Last Dev Mod (4th Nov)

**Describe the bug**
Sink rate alarms is going on loop from the departing gate till the whole flight

**To Reproduce**
1. Start a flight
2. Turn on Batteries and External Power
3.

**Expected behavior**
No Sink rate message

**Actual behavior**
Sink rate alert is going on loop

**References**

**Additional context**

Was working fine on 30th October Dev Mod

Is this a problem in the vanilla unmodded game? NO

Discord username (if different from GitHub): David Funda

### PR DESCRIPTION
**Pull request by lukecologne**
Fixes #1756, Fixes #1734, Fixes #1335

## Summary of Changes

- [x] Implement correct GPWS logic
    - [x] Implement GPWS modes
    - [x] GPWS overhead buttons now all have functionality (except for the TERR SYS button)
    - [ ] Implement GPWS self test (without terrain display test, can't do that at the moment)
- [x] Implement some sort of sound manager, so sounds don't overlap (also for minimums, altitude calls etc.)
- [x] Move callouts over to new soundmanager
   - [x] Move altitude callouts
   - [x] Move GPWS callouts
   - [x] Redo retard logic: Retard is now emitted below 20ft RA, while thrust levers not in idle (or 10ft with AP engaged).
          It is now possible to get multiple retard callouts, or no callout if you've put the thrust lever in idle before 20ft.

* Some sounds like "glideslope" or "terrain" are not in yet, but the logic is available
* Mode 2 is disabled until a fix is found (see comments)

## References

Mode 1:
grafik
Mode 2:
grafik
grafik
Mode 3:
grafik
Mode 4:
grafik
grafik
Mode 5:
grafik

grafik

Overhead controls:
grafik

Button for old GPWS:
grafik
Button for new GPWS (this is apparently in all neos):
grafik
grafik

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by lukecologne** [comment]
Alright, I think this should be ready for review. I haven't done a full test flight yet, but I tested the GPWS in specific testing conditions, and it looks like it works nicely. Pretty much all basic GPWS (no EGPWS) logic has been implemented, some sounds like "glideslope" or "terrain" are not in yet, but the logic is available. Also, GPWS self test button logic is not implemented yet, but this could be done in a later PR.

**Comment by Benjozork** [comment]
re. #1734

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by Carter633** [comment]
Name: British Avgeek YT --- Carter#1436
Date of testing: Current date 08/11/2020
Version of the sim: 1.0.8.1
PR Tested: #1742 
Object of test: GPWS Rework testing

Testing technique: Full Flight from C&D - EGCC to EGLL on VATSIM
Test streamed here:  https://youtu.be/YqqiF7IpEaU

Steps you've made to test the Pull Request:

1. Cold and dark start up, push and start
2. During Taxi - ENG1 failed and wouldn't switch back on until Xfeed and APU were on
3. TakeOff - Too Low Callouts from GPWS activated until 500ft or so
4. Cruise - Descent were fine
5. Normal ILS approach tested, Callouts and retard operated fine and as expected:

Issues (If Any):
Overall Rating: To be improved 
Conclusions of the testing: As mentioned above, suddenly with this PR, Engine 1 shut down itself during taxi - wouldn't restart until Xfeed and APU were on. Also as mentioned, during takeoff and initial climb I got the too low warnings on GPWS.

**Comment by Carter633** [comment]
Name: British Avgeek YT --- Carter#1436
Date of testing: Current date 09/11/2020
Version of the sim: 1.0.8.1
PR Tested: #1742
Object of test: GPWS Rework testing

Testing technique: Full Flight from C&D - EGKK to EGNM , turnaround then EGNM to EGLL all in one go live on VATSIM
Test streamed here: Name: British Avgeek YT --- Carter#1436
Date of testing: Current date 08/11/2020
Version of the sim: 1.0.8.1
PR Tested: #1742
Object of test: GPWS Rework testing

Testing technique: Full Flight from C&D - EGCC to EGLL on VATSIM
Test streamed here: https://youtu.be/YqqiF7IpEaU
(Half way through for landing GPWS fault)

Steps you've made to test the Pull Request:

1.Cold and dark start up, push and start
2.During Taxi - no issues
3.TakeOff - out of EGKK no issues but out of EGNM I got terrain warnings for about 1 min
4.Cruise / Descent were fine
5. Normal ILS approach tested, terrain warning activated at 1500ft and would not stop - for the return, I actually deactived it to stop but on takeoff from Leeds got too low warning on departure instead.

Issues (If Any): Got some erroneous behaviour as mentioned above - EGNM has an approach with high terrain so wanted to test it's response here, but got Terrain warnings all the way down to 50ft and no Minimums call because of that.
After parking, resetting MCDU and then departing EGNM for EGLL I switched Terrain GPWS off on the overhead, but then on initial climb got "Too Low" so turned the entire system off

Overall Rating: To be improved
Conclusions of the testing: As mentioned above, suddenly with this PR, Engine 1 shut down itself during taxi - wouldn't restart until Xfeed and APU were on. Also as mentioned, during takeoff and initial climb I got the too low warnings on GPWS.

**Comment by lukecologne** [comment]
I disabled the GPWS mode 2, which gave way too many wrong warnings. These are caused by bad / very jumpy radio altimeter simvar, which caused the derivative to be very large (which is wrong), and this caused the wrong warnings. Until this can be resolved (in a later PR), it will stay disabled.

File: A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js [hunk 1]
```diff
@@ -1,51 +1,99 @@
...
+            const FlapPosition = SimVar.GetSimVarValue("FLAPS HANDLE INDEX", "Number");
+            const FlapsInLandingConfig = FlapPushButton ? (FlapPosition === 3) : (FlapPosition === 4);
+            const vSpeed = Simplane.getVerticalSpeed();
+            const Airspeed = SimVar.GetSimVarValue("AIRSPEED INDICATED", "Knots");
+            const gearExtended = SimVar.GetSimVarValue("GEAR TOTAL PCT EXTENDED", "Percent") > 0.9;
+
+            this.update_maxRA(radioAlt);
+
+            this.GPWSMode1(radioAlt, vSpeed);
+            //this.GPWSMode2(radioAlt, Airspeed, FlapsInLandingConfig, gearExtended);
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:76]: We should add a comment saying why this is commented out.

Comment (lukecologne) [reply A32NX/html_ui/Pages/A32NX_Core/A32NX_GPWS.js:76]: True. Will add it.

**Pull request merged**